test/Spec.hs

```diff
@@ -264,6 +295,55 @@ dateBuilder m expected tod =
 matchBuilder :: Text -> Builder -> Assertion
 matchBuilder a b = LText.toStrict (Builder.toLazyText b) @?= a
 
+propWithinInsideInterval :: TimeInterval -> Bool
+propWithinInsideInterval ti@(TimeInterval t0 t1) = C.within t1 ti && C.within t0 ti
+
+propWithinOutsideInterval :: RelatedTimes -> Bool
```

Comment:
I'd prefer to avoid all these discards. This can make tests pretty inefficient. Usually I just like to make generators that generate data that doesn't need to be discarded.